apps/prairielearn/src/lib/workspace.js

```diff
@@ -59,6 +59,28 @@ module.exports = {
     workspaceUtils.init(socketServer.io);
     socketServer.io
       .of(workspaceUtils.WORKSPACE_SOCKET_NAMESPACE)
+      .use((socket, next) => {
+        // TODO: these checks are temporarily disabled until we are reasonably
+        // sure that almost all clients are sending auth information.
```

Comment:
It's possible that sticky load balancer sessions will mean that we don't have to care about this in practice; I was erring on the side of being overly cautious here. The intention is that after deploying this code, we'll wait a short amount of time (a day?) and then enable these checks.

In practice, if something goes wrong, it should be fixed by a refresh, and I think students are super familiar with refreshing if workspaces aren't working exactly right.

The downside here: since the auth/ID information are associated with the socket, we wouldn't be able to multiplex connections for multiple workspaces/variants/server-jobs over a single socket. We don't currently do this, and I can't imagine why we would in the future. Just calling it out for consideration.